lib/ansible/modules/database/influxdb/influxdb_user.py

```diff
@@ -95,6 +113,9 @@ def find_user(module, client, user_name):
                 break
     except ansible.module_utils.urls.ConnectionError as e:
         module.fail_json(msg=str(e))
+    except influx.exceptions.InfluxDBClientError as e:
+        if not (e.code == 401 and "create admin user first" in e.message):
+            module.fail_json(msg=str(e))
```

Comment:
Please do `to_native(e)` (`from ansible.module_utils._text import to_native`) instead of `str(e)`.

Also, out of curiosity, why do we ignore `(e.code == 401 and "create admin user first" in e.message)` exception?

Reply:
We ignore this exception in order to be able to create the first admin user in case authentication is activated.
In that case, influxdb will refuse all queries except the creation of an admin user, so when the module tries to check if the user already exists, it fails.

Reply:
> We ignore this exception in order to be able to create the first admin user in case authentication is activated.
> In that case, influxdb will refuse all queries except the creation of an admin user, so when the module tries to check if the user already exists, it fails.

True, but ignoring it will prevent to know already existing users and will force users to be aware that they will need to use this module in two pass :  
- first to only create an admin user
- second to create other users
I personally use the shell module to create the admin user win influx CLI (with the creation of a file to not redo it if it was already done) then use the user created with this module to add the other users

just my 2cts ^^

Reply:
Hello, thanks for the review.

I haven't used InfluxDB in a while, but I don't think I am restricting the possibilities of the users : the case you describe is only in case you enable authentication before creating any user.
In that case, according to the documentation, InfluxDB will only accept a request which creates an admin user.

Am I missing something ?

Reply:
you're right, but the first action done in the module before the creation of a user is a lookup for the already existing users and is not allowed without admin privilege when authentication is enable and will result with a "failed"
https://github.com/ansible/ansible/blob/9534fde0244bb8d1115f16d9ff9332aae574bf69/lib/ansible/modules/database/influxdb/influxdb_user.py#L216
Maybe i miss something, but as stated by @mkrizek , i don't get why you want to ignore some errors and think it is better to let as it was. 

- If you don't activate the authentication before creating the users, then you create some users (at least one admin) and will use eventually the admin user later if you activate the authentication. No need of the exception regarding 401 / create admin first
- If you activate the authentication before the users creation, you just need to create an admin user trough the CLI ([shell + creates](https://github.com/rockandska/ansible-role-influxdb/blob/master/tasks/config_management_user.yml)) to avoid the action to be trigger in later execution, then call the user module with the admin previously created. No need of the exception regarding 401 / create admin first

If you wanna update this PR, maybe some unwanted behavior are still present, but i've updated your code in [my role](https://github.com/rockandska/ansible-role-influxdb/blob/master/library/influxdb_user.py) to :  
- avoid errors when removing "ALL" privileges due the fact that the API return "ALL PRIVILEGES" instead of "ALL3 [here](https://github.com/rockandska/ansible-role-influxdb/blob/master/library/influxdb_user.py#L172-L174)
- avoid errors when a user was grant with privileges then those privileges removed then adding back some privileges again since "NO PRIVILEGES" is returnrf instead of "nothing" compare to when the user was never grant with privileges [here](https://github.com/rockandska/ansible-role-influxdb/blob/master/library/influxdb_user.py#L175-L176)
- remove privileges to the user if none was provided with the module (since byd efault user have no privilege when created) [here](https://github.com/rockandska/ansible-role-influxdb/blob/master/library/influxdb_user.py#L252-L254)
- allow privileges update even if the user already existed by removing the too early exit [here](https://github.com/ansible/ansible/blob/9534fde0244bb8d1115f16d9ff9332aae574bf69/lib/ansible/modules/database/influxdb/influxdb_user.py#L236)

You could compare the actual modifications [here](https://www.diffnow.com/files/__edp_output__2e727e56-7a5c-484e-bdb5-4f3fb2f617ef.htm) (not sure how many times the link will be available

Regards,

Ps: thanks for the code in the PR :)